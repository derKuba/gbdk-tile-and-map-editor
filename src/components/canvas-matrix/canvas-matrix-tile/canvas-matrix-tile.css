.gb-green {
  background-color: #6b790e;
}

.gb-grey {
  background-color: #c5bfbf;
}

.gb-grey-dark {
  background-color: #5a5760;
}

.gb-white {
  background-color: #fff;
}

.gb-black {
  background-color: #000;
}

.canvas-matrix-tile {
  width: 50px;
  height: 50px;

  border-right: 1px solid black;
  border-bottom: 1px solid black;
}

.grid-off {
  border: 0;
  width: 51px;
  height: 51px;
}