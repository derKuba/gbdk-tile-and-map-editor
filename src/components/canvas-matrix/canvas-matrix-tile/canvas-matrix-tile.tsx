import { Component, h, Prop } from '@stencil/core';

@Component({
    tag: 'mvjp-canvas-matrix-tile',
    shadow: false,
    styleUrl: 'canvas-matrix-tile.css',
})
export class CanvasMatrixTile {
    private colors: string[] = ['white', 'grey', 'grey-dark', 'black'];

    /**
     *   redirect click event to parent
     */
    @Prop() handleClick: Function;

    /**
     *   {x, y} coordinates
     */
    @Prop() position: object;

    /**
     *   color code of this tile
     */
    @Prop() colour: number;

    /**
     *   show borders
     */
    @Prop() showGrid: boolean;

    render() {
        const showGridclass = this.showGrid ? '' : 'grid-off';
        const colorClass = `gb-${this.colors[this.colour]}`;

        return (
            <div
                class={`canvas-matrix-tile ${colorClass} ${showGridclass}`}
                onClick={() => this.handleClick(this.position)}
                data-pos={`${JSON.stringify(this.position)}`}>
                {' '}
                <slot />
            </div>
        );
    }
}
