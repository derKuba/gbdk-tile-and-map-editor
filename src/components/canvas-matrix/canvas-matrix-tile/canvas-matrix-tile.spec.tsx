import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { CanvasMatrixTile } from './canvas-matrix-tile';

describe('mvjp-color-chooser', () => {
    it('builds', () => {
        expect(new CanvasMatrixTile()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const position = { x: 0, y: 0 };
        const onClick = () => {};
        const colourCode = 0;

        const page = await newSpecPage({
            components: [CanvasMatrixTile],
            template: () => {
                return (
                    <div>
                        <mvjp-canvas-matrix-tile
                            position={position}
                            handleClick={onClick}
                            colour={colourCode}
                            showGrid={true}>
                            here a children
                        </mvjp-canvas-matrix-tile>
                        <mvjp-canvas-matrix-tile
                            position={position}
                            handleClick={onClick}
                            colour={colourCode}
                            showGrid={false}>
                            here a children
                        </mvjp-canvas-matrix-tile>
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('handle click event', async () => {
        const position = { x: 0, y: 0 };
        const onClick = jest.fn();
        const colourCode = 0;

        const page = await newSpecPage({
            components: [CanvasMatrixTile],
            template: () => {
                return (
                    <mvjp-canvas-matrix-tile
                        position={position}
                        handleClick={onClick}
                        colour={colourCode}
                        showGrid={true}>
                        here a children
                    </mvjp-canvas-matrix-tile>
                );
            },
        });

        let button = page.root.querySelector('.canvas-matrix-tile');
        button.click();
        await page.waitForChanges();

        expect(onClick).toBeCalledWith({ x: 0, y: 0 });
    });
});
