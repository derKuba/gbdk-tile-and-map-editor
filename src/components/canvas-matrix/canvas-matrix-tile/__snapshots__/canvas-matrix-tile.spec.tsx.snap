// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-color-chooser matches snapshot 1`] = `
<mvjp-canvas-matrix-tile>
  <!---->
  <div class="canvas-matrix-tile gb-white" data-pos="{&quot;x&quot;:0,&quot;y&quot;:0}">
    here a children
  </div>
</mvjp-canvas-matrix-tile>
`;
